lib/api/model/events.dart

```diff
@@ -341,7 +341,7 @@ class UpdateMessageEvent extends Event {
   final bool? renderingOnly; // TODO(server-5)
   final int messageId;
   final List<int> messageIds;
-  final List<String> flags; // TODO enum
+  final Set<Flag> flags;
```

Comment:
model: Make `flags` on Message and UpdateMessageEvent a Set instead of List
    
    This will enable more efficient checking of whether a message has a
    certain flag.

I think this is actually a case where a `List` is likely to be more efficient than a `Set`.

The reason is that these lists will always be short, and almost always be very short. There are 7 different flags now; the list might grow occasionally but will probably never be much more than 10; so the list will never be longer than that. And most of the flags will apply to a small fraction of messages; probably over 90% of the lists of flags will be either just `[Flag.read]`, or empty.

When the list/set is small, checking membership by just looping through wins for speed because of its simplicity — there's just so little work to do. In particular when the length is 0 or 1, the loop is basically impossible to beat.

Reply:
If we wanted to really optimize this — for speed and also for space — the way to do it would be with a single int that contains flag bits. That might actually turn out to be worthwhile, given that messages are something we can have a lot of. But we can leave it out at this stage.

Reply:
Interesting. Sure, we can keep it as a List.